CardanoSharp.Wallet/Extensions/Models/CBORExtensions.cs

```diff
@@ -65,12 +63,58 @@ public static object DecodeValueByCborType(this CBORObject cborObject)
                     break;
                 case CBORType.FloatingPoint:
                     throw new NotImplementedException();
-                    break;
                 default:
                     break;
             }
 
             return result;
         }
+
+        public static short DecodeValueToInt16(this CBORObject cborObject)
+        {
+            if (cborObject.Type != CBORType.Integer)
+                throw new ArgumentException("CBORObject must be of type integer", nameof(cborObject));
+            var number = cborObject.AsNumber();
+            return number.ToInt16Checked();
+        }
+
+        public static ushort DecodeValueToUInt16(this CBORObject cborObject)
+        {
+            if (cborObject.Type != CBORType.Integer)
+                throw new ArgumentException("CBORObject must be of type integer", nameof(cborObject));
+            var number = cborObject.AsNumber();
+            return number.ToUInt16Checked();
+        }
+        public static int DecodeValueToInt32(this CBORObject cborObject)
+        {
+            if (cborObject.Type != CBORType.Integer)
+                throw new ArgumentException("CBORObject must be of type integer", nameof(cborObject));
+            var number = cborObject.AsNumber();
+            return number.ToInt32Checked();
+        }
+
+        public static uint DecodeValueToUInt32(this CBORObject cborObject)
+        {
+            if (cborObject.Type != CBORType.Integer)
+                throw new ArgumentException("CBORObject must be of type integer", nameof(cborObject));
+            var number = cborObject.AsNumber();
+            return number.ToUInt32Checked();
+        }
+
+        public static long DecodeValueToInt64(this CBORObject cborObject)
+        {
+            if (cborObject.Type != CBORType.Integer)
+                throw new ArgumentException("CBORObject must be of type integer", nameof(cborObject));
+            var number = cborObject.AsNumber();
+            return number.ToInt64Checked();
+        }
+
+        public static ulong DecodeValueToUInt64(this CBORObject cborObject)
+        {
+            if (cborObject.Type != CBORType.Integer)
+                throw new ArgumentException("CBORObject must be of type integer", nameof(cborObject));
+            var number = cborObject.AsNumber();
+            return number.ToUInt64Checked();
+        }
```

Comment:
Would this be better served from `CardanoSharp.CBOR`?  If yes, doesnt need to be now. im mainly curious

Reply:
Yeah, I would be inclined to put all the CBOR-related extension classes under CardanoSharp.Wallet.Extensions.Models under that too. 

![image](https://user-images.githubusercontent.com/67933144/172804291-95ede3b1-7016-4807-8feb-d3224f6a0679.png)